routers/private/hook_verification.go

```diff
@@ -95,6 +96,7 @@ func readAndVerifyCommit(sha string, repo *git.Repository, env []string) error {
 				if err != nil {
 					return err
 				}
+				ctx = cache.WithCacheContext(ctx)
```

Comment:
The design of `WithCacheContext` is wrong.

It will return a out-scope context and lose the current "deadline" or "cancel" control.

Before a correct design, it should never be used anymore.